### ISSUE TITLE
aminet links and amitools on Debian and Ubuntu

### ISSUE THREAD

**Issue by acf** [bebbo/amiga-gcc#382]
Hello!

Having given up trying to get this working directly on macOS I have installed UTM and a couple of linux VMs to see if I can get a working system there without fighting homebrew and Xcode :-)

I started with an x86 ubuntu system (very slow on apple silicon) and ended up with a minimal arm64 Debian, which is now nice and speedy.

1 thing I had to manually tweak is changing the various http aminet links in the makefile to https.  I have noticed occasionally when clicking on something ion the aminet website that an http link doesnt work and I have to manually change to https.  Not sure if it's something on my Mac or something at their end, but either way I just thought I'd mention that changing them to https links worked fine and maybe it's something affecting other people.

The second thing is that I have not yet been able to run make check.

amitools fails to install for me with what appears to be python errors.  I'm not too familiar with python but could it be a python2 vs python3 thing ?  

On Ubuntu it all seems happy and was clear "this is not a pip error" 

On Debian I get a complaint when installing via pip that it was "unmanaged" and it suggested I install pipx, which I did.

In both cases I ended up with same errors ( sample screenshots attached because I haven't yet figured out how best to get files out of the VM and into Mac :-).

<img width="805" alt="Screenshot 2024-02-03 at 12 16 53" src="https://github.com/bebbo/amiga-gcc/assets/218/c6ef7ad4-2f21-469e-b576-9da4b5be34e6">

<img width="805" alt="Screenshot 2024-02-03 at 12 17 10" src="https://github.com/bebbo/amiga-gcc/assets/218/0beeb7b3-608b-4203-a781-d0d61e8deb7a">

**Comment by acf** [comment]
Just to be clear, as far as I can tell the toolchain installed fine.  I just was trying to use the `make check` instructions rather than make a little sample myself.

**Comment by bebbo** [comment]
Thank you for the feedback regarding the aminet links, I'll hava a look.

Your problem with amitools... I use
```
git clone https://github.com/bebbo/amitools
cd amitools
sudo python3 setup.py install
```
to install it. And you should provide a `~/.vamosrc` containing at least
```
[volumes]
ram=/tmp
```
Also install `dejagnu` and copy the baseboards to `/usr/share/dejagnu/baseboards/`
Now `make check` should work

**Comment by acf** [comment]
@bebbo Thanks!  Will try the alternate method just now.

**Comment by acf** [comment]
Nope.  Debian still insisting that I'm not allowed to do anything in an "externally managed" environment.  setup.py complains about cython and installing cython with pipx is clearly putting it in a place that setup.py doesnt recognise.   I'll keep fiddling, but this is basically a debian/python issue.  I just thought the errors in amitools:

"Cannot assign type 'void (void) except *'to 'void (*) (void) noexcept'
• Exception values are incompatible. Suggest adding 'noexcept to type void (void) except *"

Seemed to be syntax rather than environment.

It's absolutely not a big deal.  I made a little hello.c and all was well :-)

**Comment by acf** [comment]
OK got an environment set up with python `venv` and still getting same errors using setup.py so it does feel like maybe some conflict with the cython version ?

<img width="1231" alt="Screenshot 2024-02-03 at 14 33 04" src="https://github.com/bebbo/amiga-gcc/assets/218/ea6cc5b3-596d-400a-aeb6-4abe7297ea42">

**Comment by bebbo** [comment]
installing via pip should work now

**Comment by bebbo** [comment]
You may also use the original repo once this issue https://github.com/cnvogelg/machine68k/issues/1 is resolved.
